### DISCUSSION TITLE
CCbutton Help

### DISCUSSION THREAD

**Discussion by Darkluiside305** [tttapa/Control-Surface#618] [Q&A]
IM trying to add CCbuttons with the CCPotentiometer but idk if im doing it right or not, when i add a second one 

```cpp
#include <Arduino.h>

#include <Control_Surface.h> // Include the Control Surface library
 
// Instantiate a MIDI over USB interface.
USBMIDI_Interface midi;
 
// Instantiate an array of CCPotentiometer objects
CCPotentiometer potentiometers[] {
  //{A0,0x10},
  //{A1,0x11},
  //{A3,0x13},
  //{A4,0x14},
  //{A5,0x15},
  //{A6,0x16},
  //{A2,0x17},
  //{A2,0x18},
  //{A2,0x19},
  //{A10,0x1A},



};

CCButton button = {
  // Push button on pin 5:
  5

,
  // General Purpose Controller #1 on MIDI channel 1:
  
  {MIDI_CC::General_Purpose_Controller_1, CHANNEL_1},

  6,

  {MIDI_CC::General_Purpose_Controller_1, CHANNEL_2},

};



void setup() {
  Control_Surface.begin(); // Initialize Control Surface
}
 
void loop() {
  Control_Surface.loop(); // Update the Control Surface
}
```

this is the error i get

![Screenshot_2](https://user-images.githubusercontent.com/47836134/140724434-c698bbf8-4050-4b19-9d95-89f2bef4b3e7.png)

im a noob on this any help with be appreciated

=)

**Comment by Veroledez** [comment]
Hola, creo que es asi


CCButton button = {

{5,   {MIDI_CC :: General_Purpose_Controller_1, CHANNEL_1}},

{6,   {MIDI_CC :: General_Purpose_Controller_1, CHANNEL_2}},

};


El pin y la direccion midi en un {}

**Comment by Darkluiside305** [comment]
Do i  put it on the same spot?

Lo pongo en el mismo citio?

**Comment by Veroledez** [comment]
Lo que te puse arriba reemplazalo en el codigo que subiste

**Comment by Darkluiside305** [comment]
ahh, i did that and still get the same error

trate, pero todabia tengo el mismo error
![254159011_1026230651506606_2560722599279367140_n](https://user-images.githubusercontent.com/47836134/140844368-cee7846d-a021-42bd-8f9b-a35791c23baa.png)

**Reply by tttapa** [reply]
See https://www.learncpp.com/cpp-tutorial/arrays-part-i/.

```cpp
CCButton button[] {
  {5, {MIDI_CC::General_Purpose_Controller_1, CHANNEL_1} },
  {6, {MIDI_CC::General_Purpose_Controller_1, CHANNEL_2} },
};
```

**Reply by Darkluiside305** [reply]
that worked perfectly, thanx bro
